File: backend/app/domain/entities/resultado_simulacion_activo_generacion.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class ResultadoSimulacionActivoGeneracionEntity:
    idResultadoActivoGen: Optional[int] = None
    energiaTotalGenerada_kWh: Optional[float] = None
    factorCapacidad_pct: Optional[float] = None
    performanceRatio_pct: Optional[float] = None
    horasOperacionEquivalentes: Optional[float] = None
    idResultadoSimulacion: Optional[int] = None
    idActivoGeneracion: Optional[int] = None

File: frontend/lib/models/resultado_simulacion_activo_generacion.dart
import 'package:json_annotation/json_annotation.dart';

part 'resultado_simulacion_activo_generacion.g.dart';

@JsonSerializable()
class ResultadoSimulacionActivoGeneracion {
  final int? idResultadoActivoGen;
  final double? energiaTotalGenerada_kWh;
  final double? factorCapacidad_pct;
  final double? performanceRatio_pct;
  final double? horasOperacionEquivalentes;
  final int? idResultadoSimulacion;
  final int? idActivoGeneracion;

  ResultadoSimulacionActivoGeneracion({
    this.idResultadoActivoGen,
    this.energiaTotalGenerada_kWh,
    this.factorCapacidad_pct,
    this.performanceRatio_pct,
    this.horasOperacionEquivalentes,
    this.idResultadoSimulacion,
    this.idActivoGeneracion,
  });

  factory ResultadoSimulacionActivoGeneracion.fromJson(Map<String, dynamic> json) => _$ResultadoSimulacionActivoGeneracionFromJson(json);
  Map<String, dynamic> toJson() => _$ResultadoSimulacionActivoGeneracionToJson(this);
}
